service/raspbian-install-device-agent.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+if [[ $EUID -ne 0 ]]; then
+    echo "This script must be run as root"
+    exit 1
+fi
+
+# Node.js version
+MIN_NODEJS=16
+
+# Update package list and upgrade installed packages
+apt-get update
+
+# Helper functions to test for existence of npm
+function HAS_NPM {
+    if [ -x "$(command -v npm)" ]; then return 0; else return 1; fi
+}
+# Install Node.js and npm if they do not exist
+if [ -x "$(command -v node)" ]; then
+    echo "NodeJS found"
+    VERSION=$(node --version | cut -d "." -f1 | cut -d "v" -f2)
+
+    if [[ $VERSION -ge $MIN_NODEJS ]]; then
+        echo "**************************************************************"
+        echo " NodeJS Version $MIN_NODEJS or newer found "
+        echo "**************************************************************"
+    else
+        echo "**************************************************************"
+        echo " You need NodeJS $MIN_NODEJS or newer, please upgrade "
+        echo "**************************************************************"
+        exit 1
+    fi
+else
+    echo "**************************************************************"
+    echo " No NodeJS found"
+    echo " Do you want to install NodeJS $MIN_NODEJS?"
+    echo "**************************************************************"
+    read -p "y/N: " yn
+    [ -z "$yn" ] && yn="n"
+
+    if [[ "$yn" == "y" ]] || [[ "$yn" == "Y" ]]; then
+        if [ ! -x "$(command -v curl)" ]; then
+            sudo apt-get install -y curl
+        fi
+        sudo apt-get install -y ca-certificates curl gnupg
+        sudo mkdir -p /etc/apt/keyrings
+        curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | sudo gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
+        echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$MIN_NODEJS.x nodistro main" | sudo tee /etc/apt/sources.list.d/nodesource.list
+        sudo apt-get update
+        sudo apt-get install nodejs -y
+    else
+        echo "**************************************************************"
+        echo " You will need to manually install NodeJS first. "
+        echo " Exiting."
+        echo "**************************************************************"
+        exit 1
+    fi
+fi
```

Comment:
This won't work on ARMv6 builds as the standard NodeJS builds don't support it. We should document that Pi Zero's are not supported

Reply:
I will add it to the doc: https://flowfuse.com/docs/hardware/introduction/